include/zephyr/bluetooth/l2cap.h

```diff
@@ -557,6 +558,52 @@ int bt_l2cap_ecred_chan_connect(struct bt_conn *conn,
  */
 int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu);
 
+/** @brief Reconfigure Enhanced Credit Based L2CAP channels
+ *
+ *  Experimental API to reconfigure L2CAP ECRED channels with explicit MPS and
+ *  MTU values.
+ *
+ *  Pend a L2CAP ECRED reconfiguration for up to 5 channels. All provided
+ *  channels must share the same @ref bt_conn.
+ *
+ *  Cannot decrease MTU of any channel. And, cannot decrease MPS of any channel
+ *  when more than one channel is provided.
+ *
+ *  There is no dedicated callback for this operation, but whenever a peer responds to a
+ *  reconfiguration request, each channel's reconfigured() callback is invoked.
+ *
+ *  This function may block for the duration of L2CAP_RTX_TIMEOUT.
+ *
+ *  @warning Known issue: The implementation returns -EBUSY if there is a
+ *  concurrent reconfigure operation on the same connection. The caller may try
```

Comment:
Not sure I understand; why is this an issue? Isn't it just a fact that we can only have 1 outstanding reconfig request at any one time?

Reply:
The issue is that there is no event for when the reconfigure resource becomes available. I'll reword.